### PR TITLE
test: add test for creating new dhis2 instance

### PR DESCRIPTION
**Pull request by radnov**
Related to https://github.com/dhis2-sre/im-manager/pull/625

This PR slightly updates the playwright workflow, so that it can be triggered manually with a `workflow_dispatch` event and adds a new e2e test for creating a DHIS2 instance. Note that the test waits for the instance to be ready (aka `Running`) and then deletes it.

Note that triggering the PR from the current repository will target the dev API, while triggering it "remotely" from the `im-manager` repo will target the API passed by the `im-manager` calling workflow.

p.s. Currently the "create new dhis2 instance" test won't make sure that the instance is ready after it's created, as we don't show that status in the UI. The test will be updated once the status is shown (https://dhis2.atlassian.net/browse/DEVOPS-351).

### CODE REVIEW

File: e2e/utils/upload-database.ts
Comment (HendrikThePendric) [inline e2e/utils/upload-database.ts:14]: I hope that what I am about to write is of any use, because it's going to be a bit of a StackOverflow style answer, not at all providing a solution but rather redefining the problem....

1. Ideally we start with a clean backend, so we always _have to_ create this test database at the start of the run. And since the backend is pristine, we _know_ that it hasn't been created yet. If this was the case, we didn't have to have any conditional logic in place.
2. Uploading the test database is not currently part of the test suite. And even it if was to become part of it, we shouldn't rely on the side effects of one test in other tests. As such I'd say this is more part of the test suite's setup process. So we need to probably do this in one of the ways outlined in the [Global setup and teardown docs](https://playwright.dev/docs/test-global-setup-teardown).
3. Since this is not part of the test suite itself, I guess it could be cleaner to upload the file using a HTTP request. However, doing it via the UI seems to be very acceptable too going by the docs.

**Review by HendrikThePendric — CHANGES_REQUESTED** [review]
I've left a few comments regarding the test steps.

File: e2e/instances.spec.ts
Comment (radnov) [reply e2e/instances.spec.ts:47]: I tried something similar, but I forgot to mention one small but very important detail - it's required to reload the page in order to see a change in the status. 👀  Therefore, this doesn't really work, unless we can add a page reload in there somehow? 🤷

File: e2e/instances.spec.ts
Comment (radnov) [reply e2e/instances.spec.ts:55]: Agreed 👍

File: e2e/instances.spec.ts
Comment (radnov) [reply e2e/instances.spec.ts:29]: Agreed 👍

File: e2e/utils/upload-database.ts
Comment (radnov) [reply e2e/utils/upload-database.ts:14]: So in summary I suppose the way to go here would be to upload the database as part of the global [Setup](https://playwright.dev/docs/test-global-setup-teardown#setup) and then delete it as part of the global [Teardown](https://playwright.dev/docs/test-global-setup-teardown#setup), right? 🤔 

@tonsV2 Any thoughts on this?

File: e2e/instances.spec.ts
Comment (HendrikThePendric) [reply e2e/instances.spec.ts:46]: Aha one of those cases where we can clean up the test after we've cleaned up the code 🙄 

First a practical suggestion, to reload the window, just call `await page.reload()` ([see docs](https://playwright.dev/docs/api/class-page#page-reload))

An on a more general note.... Since this is a single-page-application we can't really say a feature is fully functional if it requires a full page reload. So technically this is functionality is broken, and having a test in place to assert the broken functionality is working as expected but not as required feels rather odd.

File: e2e/utils/upload-database.ts
Comment (HendrikThePendric) [reply e2e/utils/upload-database.ts:14]: What you're suggesting _could_ be the right approach. But it depends: 
1. If we need this DB for one test only, then I would actually recommend using the `beforeAll` and `afterAll` hooks of the test that requires the DB (see [docs](https://playwright.dev/docs/api/class-test#test-before-all-1)).
2. If we (expect to) need this DB for more tests, then using the global setup and teardown seems more appropriate.

Comment (radnov) [reply e2e/utils/upload-database.ts:14]: Alright, will stick with `beforeAll` and `afterAll` for now. 👍

File: e2e/instances.spec.ts
Comment (radnov) [reply e2e/instances.spec.ts:46]: > First a practical suggestion, to reload the window, just call `await page.reload()`

Yeah, that's what I'm doing already. 👍 

> An on a more general note.... Since this is a single-page-application we can't really say a feature is fully functional if it requires a full page reload. So technically this is functionality is broken, and having a test in place to assert the broken functionality is working as expected but not as required feels rather odd.

At this point, with the instances list view getting deprecated anytime soon, I'm not sure how to continue here. Should we merge this as is and then fix the tests after the UI changes (to deployments list view) or should I wait until the new UI changes are merged and then resume working on this?

Adding @tonsV2 in the loop.

Comment (tonsV2) [reply e2e/instances.spec.ts:47]: Branches have a tendency to deprecate unless merged, so unless you really want to save this for later I'd say go ahead and merge. Just reload the page for now and let's update the test later. Please make a jira and a also comment in the code about the shortcomings.

**Review by tonsV2 — APPROVED** [review]

File: e2e/instances.spec.ts
Comment (radnov) [reply e2e/instances.spec.ts]: I have removed the instance status check for now and added a comment in the code, as well as the following Jira task to keep track of this - https://dhis2.atlassian.net/browse/DEVOPS-351

**Review by HendrikThePendric — APPROVED** [review]
The follow-up work is documented, so I agree that this is ready for merging.

**Pull request merged**
